### DISCUSSION TITLE
Secured Files

### DISCUSSION THREAD

**Discussion by mrwyndham** [pocketbase/pocketbase#1506] [Q&A]
Hello, I have a question about how one would be able to secure their files. Distressingly, I am unable to require authentication to download files from a collection. I simply follow the following syntax

http://127.0.0.1:8090/api/files/COLLECTION_ID_OR_NAME/RECORD_ID/FILENAME 

and regardless of any rules placed on the collection I am capable of downloading that file. Am I overlooking anything?

**Comment by ganigeorgiev** [comment]
Private/Protected files are currently not supported but it is in the roadmap and will be handled in v0.13/v0.14 as part of https://github.com/pocketbase/pocketbase/issues/215.

**Comment by xrd** [comment]
@mrwyndham If you push to S3 (https://pocketbase.io/docs/files-handling/#storage-options) then you could implement your own access controls. Minio is a great open source alternative to S3. You could do whatever you want (post-processing or any form of access control). It means you are managing the access outside of pocketbase, but for some circumstances that might be preferable anyway.
